src/stronghold.hs

```diff
@@ -228,7 +236,14 @@ site runStoreOp =
     json <- runStoreOp $ do
       hier <- Trees.lastHierarchy hist
       snd <$> Trees.materializedView path hier
+
+    let etag = (Base16.encode . hash . toStrict . Aeson.encode) json
+
+    Snap.modifyResponse $ Snap.setHeader etagHeader etag
     Snap.writeLBS $ Aeson.encode json
+   where
+     -- BS.toStrict is available in more moderns versions of GHC
+     toStrict = fromMaybe BS.empty . listToMaybe . LBS.toChunks
```

Comment:
do you mean to only return the first chunk? would `BS.concat . LBS.toChunks` work?